perception/lidar_transfusion/src/lidar_transfusion_node.cpp

```diff
@@ -0,0 +1,161 @@
+  class_names_ = this->declare_parameter<std::vector<std::string>>("class_names");
+  const std::string trt_precision = this->declare_parameter("trt_precision", "fp16");
+  const auto voxels_num = this->declare_parameter<std::vector<int64_t>>("voxels_num");
+  const auto point_cloud_range = this->declare_parameter<std::vector<double>>("point_cloud_range");
+  const auto voxel_size = this->declare_parameter<std::vector<double>>("voxel_size");
+  const std::string onnx_path = this->declare_parameter<std::string>("onnx_path", "");
+  const std::string engine_path = this->declare_parameter<std::string>("engine_path", "");
+
+  // pre-process
+  const std::string densification_world_frame_id =
+    this->declare_parameter("densification_world_frame_id", "map");
+  const int densification_num_past_frames =
+    this->declare_parameter("densification_num_past_frames", 1);
+
+  // post-process
+  const float circle_nms_dist_threshold =
+    static_cast<float>(this->declare_parameter<double>("circle_nms_dist_threshold", 0.5));
+  {  // IoU NMS
+    NMSParams p;
+    p.nms_type_ = NMS_TYPE::IoU_BEV;
+    p.target_class_names_ =
+      this->declare_parameter<std::vector<std::string>>("iou_nms_target_class_names");
+    p.search_distance_2d_ = this->declare_parameter<double>("iou_nms_search_distance_2d");
+    p.iou_threshold_ = this->declare_parameter<double>("iou_nms_threshold");
+    iou_bev_nms_.setParameters(p);
+  }
+  const auto yaw_norm_thresholds =
+    this->declare_parameter<std::vector<double>>("yaw_norm_thresholds");
+  const float score_threshold =
+    static_cast<float>(this->declare_parameter<double>("score_threshold", 0.2));
+
+  NetworkParam network_param(onnx_path, engine_path, trt_precision);
+  DensificationParam densification_param(
+    densification_world_frame_id, densification_num_past_frames);
+  TransfusionConfig config(
+    voxels_num, point_cloud_range, voxel_size, circle_nms_dist_threshold, yaw_norm_thresholds,
+    score_threshold);
+
+  const auto allow_remapping_by_area_matrix =
+    this->declare_parameter<std::vector<int64_t>>("allow_remapping_by_area_matrix");
+  const auto min_area_matrix = this->declare_parameter<std::vector<double>>("min_area_matrix");
+  const auto max_area_matrix = this->declare_parameter<std::vector<double>>("max_area_matrix");
+  detection_class_remapper_.setParameters(
+    allow_remapping_by_area_matrix, min_area_matrix, max_area_matrix);
+
+  detector_ptr_ = std::make_unique<TransfusionTRT>(network_param, densification_param, config);
+
+  cloud_sub_ = this->create_subscription<sensor_msgs::msg::PointCloud2>(
+    "~/input/pointcloud", rclcpp::SensorDataQoS{}.keep_last(1),
+    std::bind(&LidarTransfusionNode::cloudCallback, this, std::placeholders::_1));
+
+  objects_pub_ = this->create_publisher<autoware_auto_perception_msgs::msg::DetectedObjects>(
+    "~/output/objects", rclcpp::QoS(1));
+
+  published_time_pub_ = std::make_unique<tier4_autoware_utils::PublishedTimePublisher>(this);
+
```

Comment:
(The comment does not apply specifically to this line)

In centerpoint there is parameter dimensionality check for the pointcloud ranges and voxel sizes.
Is there a particular reason for not doing the same in transfusion?

```
  if (point_cloud_range.size() != 6) {
    RCLCPP_WARN_STREAM(
      rclcpp::get_logger("lidar_centerpoint"),
      "The size of point_cloud_range != 6: use the default parameters.");
  }
  if (voxel_size.size() != 3) {
    RCLCPP_WARN_STREAM(
      rclcpp::get_logger("lidar_centerpoint"),
      "The size of voxel_size != 3: use the default parameters.");
  }
```

Reply:
Right, this might me helpful.
I added parameter descriptor as well since all the parameters shouldn't be modified. Unfortunately, it's not possible to set limits in arrays this way (it works only for primitive types). It means at the node level it's possible to set e.g. negative `voxel_size`. However, json schema (https://github.com/autowarefoundation/autoware.universe/pull/6890/commits/ac01c4a29a362b902c99e4a1b2d18e5ff5cb2b15) acts as a validation from Autoware design point of view.
https://github.com/autowarefoundation/autoware.universe/pull/6890/commits/8434721419a0612630c3072c46838fe6af8c5c86